### ISSUE TITLE
Doesn't support maps.

### ISSUE THREAD

**Issue by kayabaNerve** [PMunch/protobuf-nim#21]
```
message ResultMap {
  map<string, Result> results = 3;
}
```

throws an error when being parsed. As this is syntactic sugar for:

```
message MapFieldEntry {
  key_type key = 1;
  value_type value = 2;
}

repeated MapFieldEntry map_field = N;
```

this isn't that big of a deal. That said, it is part of the Protobuf 3 spec.

**Comment by PMunch** [comment]
I know. To speed up development of the library I opted to not implement maps as they can be defined by a normal message as you pointed out. I'm more than happy to accept a PR though for an implementation of it if I don't find the time to implement it myself at some point.

**Comment by kayabaNerve** [comment]
Yep. I'm actually planning on submitting some improvements soon. I not only wanted to track this, but also check with you on it. Thanks for the heads up.
